kotlinx-coroutines-core/jvm/src/debug/internal/DebugProbesImpl.kt

```diff
@@ -31,47 +31,40 @@ internal object DebugProbesImpl {
     private val installations = atomic(0)
 
     /**
-     * The current location in a coroutine that may be retrieved at a breakpoint.
-     */
-    // TODO: add tests
-    public val currentPosition: LocationInCoroutine 
-        get() = LocationInCoroutineImpl(
-            capturedCoroutines.asSequence().mapNotNull { owner ->
-                when {
-                    // a coroutine is completed
-                    owner.isFinished() -> null
-                    // a coroutine is executed in another thread
-                    owner.info.lastObservedThread != Thread.currentThread() -> null
-                    // an active coroutine that is being executed on the current thread
-                    else -> owner.info.sequenceNumber
-                }
-            }.toSet()
-        )
-
-    /**
-     * For every debug step inside the coroutine we need to determine whether
-     * we are about to stop in the same coroutine we were on the previous step.
+     * This API is used for stepping in the debugger.
      * 
-     * [LocationInCoroutine] represents a location inside a coroutine body and introduces 
-     * [canRunTo] method which returns true if a step from the current location the given location is possible.
+     * A debug step from one location to the next location within a coroutine is allowed if 
+     * the same coroutine that was running at the previous location continues its execution at the next location.
      * 
-     * The step is possible if we stay in the same coroutine or we step in an undispatched coroutine.
-     * TODO: example
+     * We also allow to step into coroutines that started its execution in the current thread immediately without being dispatched.
      * 
-     * The debugger can get the current location in a coroutine at the breakpoint and invoke [canRunTo] method to define whether we can run to another location.
+     * [CoroutinesOnThread] captures a set of coroutines that were running on the current thread at a breakpoint, 
+     * and can determine whether we can step to the current location in code by checking if any of those captured coroutines 
+     * continue its execution at the current location.
      */
-    public sealed interface LocationInCoroutine {
-        public fun canRunTo(other: LocationInCoroutine): Boolean
-        public fun allCoroutines(): String
+    public class CoroutinesOnThread internal constructor(private val coroutinesAtPreviousLocation: Set<Long>) {
+        public fun canRunToCurrentLocation(): Boolean {
```

Comment:
`canRunToCurrentLocation`  may be a property as well.

Reply:
According to https://kotlinlang.org/docs/coding-conventions.html#functions-vs-properties, only 1/3 of the criteria fit, so I think this should stay a function. It's a major surprise when the value of the field `x.y` is `true` when read on one thread but `false` when read on another one.